### PR TITLE
Updated dnat docs with changes in protocol handling

### PR DESCRIPTION
**Pull request by vbauzysvmware**
Part of issue #112
Signed-off-by: Vaidotas Bauzys <vbauzys@vmware.com>

### CODE REVIEW

File: website/docs/r/dnat.html.markdown [hunk 1]
```diff
@@ -15,21 +15,38 @@ and delete destination NATs to map an external IP/port to an internal IP/port.
 
 ```hcl
 resource "vcd_dnat" "web" {
+  org              = "my-org" #Optional
+  vdc              = "my-vdc" #Optional
```
Comment (lvirbalas) [inline website/docs/r/dnat.html.markdown]: `#Optional` -> `# Optional`

File: website/docs/r/dnat.html.markdown [hunk 2]
```diff
@@ -15,21 +15,38 @@ and delete destination NATs to map an external IP/port to an internal IP/port.
...
 
 The following arguments are supported:
-
 * `edge_gateway` - (Required) The name of the edge gateway on which to apply the DNAT
 * `external_ip` - (Required) One of the external IPs available on your Edge Gateway
-* `port` - (Required) The port number to map
+* `port` - (Required) The port number to map. -1 translates to "any"
+* `translated_port` - (Optional) The port number to map
 * `internal_ip` - (Required) The IP of the VM to map to
+* `protocol` - (Optional; *v2.0+*) The protocol type. Possible values TCP, UDP, TCPUDP, ICMP, ANY. TCP is default to be back compatible with previous version
```
Comment (lvirbalas) [inline website/docs/r/dnat.html.markdown]: Example shows lower case "icmp", but here types are upper case. Need to clarify which case to use.

File: website/docs/r/dnat.html.markdown [hunk 3]
```diff
@@ -15,21 +15,38 @@ and delete destination NATs to map an external IP/port to an internal IP/port.
...
+* `icmp_sub_type` - (Optional; *v2.0+*) The name of ICMP type.
```
Comment (lvirbalas) [inline website/docs/r/dnat.html.markdown]: Is there a supported list of values for icmp_sub_type to reference? Like in example you have "router-solicitation", where did you find it?

File: website/docs/r/dnat.html.markdown (hunk 2 above)
Comment (vbauzysvmware) [reply website/docs/r/dnat.html.markdown]: Support both types

File: website/docs/r/dnat.html.markdown (hunk 3 above)
Comment (vbauzysvmware) [reply website/docs/r/dnat.html.markdown]: There isn't explicit normal documentation for that

**Comment by vbauzysvmware** [comment]
Remarks fixed

File: website/docs/r/dnat.html.markdown [hunk 4]
```diff
@@ -15,21 +15,38 @@ and delete destination NATs to map an external IP/port to an internal IP/port.
...
 The following arguments are supported:
-
 * `edge_gateway` - (Required) The name of the edge gateway on which to apply the DNAT
 * `external_ip` - (Required) One of the external IPs available on your Edge Gateway
-* `port` - (Required) The port number to map
+* `port` - (Required) The port number to map. -1 translates to "any"
+* `translated_port` - (Optional) The port number to map
 * `internal_ip` - (Required) The IP of the VM to map to
+* `protocol` - (Optional; *v2.0+*) The protocol type. Possible values TCP, UDP, TCPUDP, ICMP, ANY. TCP is default to be back compatible with previous version
+* `icmp_sub_type` - (Optional; *v2.0+*) The name of ICMP type. Possible values address-mask-request, destination-unreachable,echo-request,echo-reply, parameter-problem, redirect, router-advertisement, router-solicitation, source-quench, time-exceeded, timestamp-request, timestamp-reply, any
```
Comment (lvirbalas) [inline website/docs/r/dnat.html.markdown]: Polish and add missing spaces after commas here: `, echo-request, echo-reply`

File: website/docs/r/dnat.html.markdown [hunk 5]
```diff
@@ -15,21 +15,38 @@ and delete destination NATs to map an external IP/port to an internal IP/port.
...
+
+resource "vcd_dnat" "forIcmp" {
+  org              = "my-org" # Optional
+  vdc              = "my-vdc" # Optional
+  
+  edge_gateway = "Edge Gateway Name"
+  external_ip  = "78.101.10.20"
+  port         = -1
+  internal_ip  = "10.10.0.5"
+  protocol = "ICMP"
```
Comment (lvirbalas) [inline website/docs/r/dnat.html.markdown]: What about adding a remark:
`protocol = "icmp" # The value of this parameter is case-insensitive`

File: website/docs/r/dnat.html.markdown (hunk 4 above)
Comment (lvirbalas) [reply website/docs/r/dnat.html.markdown]: Also may be worth adding a remark about case insensitivity:
`The name of ICMP type (the value of this parameter is case-insensitive)`

File: website/docs/r/dnat.html.markdown [hunk 6]
```diff
@@ -15,21 +15,38 @@ and delete destination NATs to map an external IP/port to an internal IP/port.
...
   translated_port = 8080
 }
+
+resource "vcd_dnat" "forIcmp" {
+  org              = "my-org" # Optional
+  vdc              = "my-vdc" # Optional
+  
+  edge_gateway = "Edge Gateway Name"
+  external_ip  = "78.101.10.20"
+  port         = -1
```
Comment (dataclouder) [inline website/docs/r/dnat.html.markdown]: Please add an inline comment saying that "-1" == "any". Even if it is in the description below, it's an odd enough value that deserves an immediate explanation

File: website/docs/r/dnat.html.markdown [hunk 7]
```diff
@@ -15,21 +15,38 @@ and delete destination NATs to map an external IP/port to an internal IP/port.
...
 
 The following arguments are supported:
-
 * `edge_gateway` - (Required) The name of the edge gateway on which to apply the DNAT
 * `external_ip` - (Required) One of the external IPs available on your Edge Gateway
-* `port` - (Required) The port number to map
+* `port` - (Required) The port number to map. -1 translates to "any"
+* `translated_port` - (Optional) The port number to map
 * `internal_ip` - (Required) The IP of the VM to map to
+* `protocol` - (Optional; *v2.0+*) The protocol type. Possible values TCP, UDP, TCPUDP, ICMP, ANY. TCP is default to be back compatible with previous version
```
Comment (dataclouder) [inline website/docs/r/dnat.html.markdown]: "Possible values TCP, ..." ==> "Possible values are TCP, ..."
"back compatible" ==> "backward compatible"

Can we add a link to the place where the protocols are defined?

File: website/docs/r/dnat.html.markdown [hunk 8]
```diff
@@ -15,21 +15,38 @@ and delete destination NATs to map an external IP/port to an internal IP/port.
...
+* `icmp_sub_type` - (Optional; *v2.0+*) The name of ICMP type. Possible values address-mask-request, destination-unreachable, echo-request, echo-reply, parameter-problem, redirect, router-advertisement, router-solicitation, source-quench, time-exceeded, timestamp-request, timestamp-reply, any
```
Comment (dataclouder) [inline website/docs/r/dnat.html.markdown]: "Possible values" ==> "Possible values are".
Is it possible to add a link to the place where these sub-types are defined/explained?

File: website/docs/r/dnat.html.markdown (hunk 7 above)
Comment (vbauzysvmware) [reply website/docs/r/dnat.html.markdown]: There isn't exact documentation for that. You can see them in VCD UI

File: website/docs/r/dnat.html.markdown (hunk 8 above)
Comment (vbauzysvmware) [reply website/docs/r/dnat.html.markdown]: There isn't exact documentation for that. You can see them in VCD UI

**Pull request merged**